solr/core/src/java/org/apache/solr/handler/ContentStreamHandlerBase.java

```diff
@@ -101,6 +112,30 @@ public void handleRequestBody(SolrQueryRequest req, SolrQueryResponse rsp) throw
     }
   }
 
+  /**
+   * Check if UPDATE circuit breakers are tripped. Override this method in sub classes that do not
+   * want to check circuit breakers.
+   *
+   * @return true if circuit breakers are tripped, false otherwise.
+   */
+  protected boolean checkCircuitBreakers(SolrQueryRequest req, SolrQueryResponse rsp) {
+    CircuitBreakerRegistry circuitBreakerRegistry = req.getCore().getCircuitBreakerRegistry();
+    if (circuitBreakerRegistry.isEnabled(RequestType.UPDATE)) {
+      List<CircuitBreaker> trippedCircuitBreakers =
+          circuitBreakerRegistry.checkTripped(RequestType.UPDATE);
+      if (trippedCircuitBreakers != null) {
+        String errorMessage = CircuitBreakerRegistry.toErrorMessage(trippedCircuitBreakers);
+        rsp.add(STATUS, FAILURE);
+        rsp.setException(
+            new SolrException(
+                SolrException.ErrorCode.SERVICE_UNAVAILABLE,
```

Comment:
Have you considered using a different error code for circuit breakers?
HTTP `429 Too Many Requests` could be appropriate here. Mostly, if we want the Solr caller to slow down, it is required to somehow expose that we're not just facing a generic failure but some self protection mechanisms.

Reply:
I agree this sounds like a more appropritate code. Perhaps it has better support in load balancers/gateways?

The existing CB for query uses 503 code, so we should perhaps not change the code in 9.x.

So we could safely change to 429 in main branch and document it as a breaking change. Or we could change it for the new pluggable CBs, but not for the legacy `CircuitBreakerManager`, providing back compat for existing users/configs. Wdyt?

Reply:
So I implemented code 429 for the new pluggable CBs, but keeping 503 for `CircuitBreakerManager` for back compat.

Reply:
Good point about compatibility of legacy `CircuitBreakerManager`. Thanks!

Reply:
Yep, we can remove `CircuitBreakerManager` along with the 503 code handling logic in main branch (in another PR). There is a deprecation notice and upgrade note as part of #96 that alerts people of it going away.